### PR TITLE
Fix resizing

### PR DESCRIPTION
**Pull request by CPBridge**
Currently the resizing (i.e. random downsampling) operations are not working for me when I pass images with a channel dimension (as specified in the docstring and as seems to work fine for the other augmentations). They are failing due to a dimension mismatch. Also the logic for the downsampling down a single axis seems wrong: the scale factor should be '1' for the non-downsampled axes but that is not the case.

There's some small chance I'm missing something perhaps but I think the changes are needed to restore the expected functionality.  I also added a check on the `voxsize` parameter as this can cause confusing error messages if the user gets it wrong.

@adalca @ahoopes

### CODE REVIEW

**Comment by adalca** [comment]
there's some downsampling happening here -- we should probably add some anti-aliasing (blurring). @ahoopes does that make sense?

**Comment by CPBridge** [comment]
Note that [torch.nn.interpolate](https://pytorch.org/docs/stable/generated/torch.nn.functional.interpolate.html) supports an `antialias` flag, but frustratingly it is not compatible with `'trilinear'` interpolation mode

**Comment by ahoopes** [comment]
thanks @CPBridge - looks great to me

and yeah, smoothing would be ideal. a lot of the resizing stuff was quickly hacked together for the initial push, so it's missing some useful features. could use voxy behind the scenes eventually to make this implementation cleaner as welll

maybe for now we should make a relatively universal interpolation function that includes antialiasing / smoothing

**Comment by ahoopes** [comment]
@CPBridge you're welcome to pull this once you think it's ready. do you have perms to do that?

**Comment by adalca** [comment]
At the very least we should mark a todo about the anti-aliasing/deblurring, first. IMO.

**Comment by CPBridge** [comment]
I added a basic anti-aliasing with gaussian blur in [a0e4cd2](https://github.com/dalcalab/voxynth/pull/6/commits/a0e4cd26f3ed19a92ce091fadcd05d0cf6cbd45b).

No I do not have permissions to merge to main, so I'll need one of you to do this please
